lib/pavilion/utils.py

```diff
@@ -193,6 +195,18 @@ def path_is_external(path: Path):
     return not_up_refs - up_refs <= 0
 
 
+def symlinktree(source_directory, destination_directory):
+    for root, dirs, files in os.walk(source_directory):
+        for file in files:
+            src_path = os.path.join(root, file)
+            rel_path = os.path.relpath(src_path, source_directory)
+            dst_path = os.path.join(destination_directory, rel_path)
+
+            # Create
+            os.makedirs(os.path.dirname(dst_path), exist_ok=True)
+            os.symlink(src_path, dst_path)
+
+
```

Comment:
I started write that you should mimic how this is done in `builder.py` with `maybe_symlink_copy`, but with some of the options you need to add this way is better.